index.js

```diff
@@ -1,138 +1,155 @@
-console.log('Loading Discovery Bib Poster');
+console.log('Loading Discovery Bib Poster')
 
-const avro = require('avsc');
-const OAuth = require('oauth');
-const Promise = require('promise');
-const request = require('request');
+const avro = require('avsc')
+const OAuth = require('oauth')
+const Promise = require('promise')
+const request = require('request')
 
 // Initialize cache
-var CACHE = {};
+var CACHE = {}
 
 // kinesis stream handler
-exports.kinesisHandler = function(records, context, callback) {
-  console.log('Processing ' + records.length + ' records');
+exports.kinesisHandler = function (records, context, callback) {
+  console.log('Processing ' + records.length + ' records')
 
   // retrieve token and schema
   Promise.all([token(), schema()])
     .then(function (res) {
-      var access_token = res[0];
-      var schema = res[1];
-      onReady(records, access_token, schema);
-    });
+      var accessToken = res[0]
+      var schema = res[1]
+      onReady(records, accessToken, schema)
+    })
 
   // run when access token and schema are loaded
-  function onReady(payload, access_token, schema) {
-    // console.log('Ready', access_token, schema);
+  function onReady (payload, accessToken, schema) {
+    console.log('Ready', accessToken, schema)
     // load avro schema
-    var avroType = avro.parse(schema);
+    var avroType = avro.parse(schema)
     // parse payload
-    var data = payload
-      .map(function(record){
-        return parseKinesis(record, avroType);
-      });
+    var records = payload
+      .map(function (record) {
+        return addSource(parseKinesis(record, avroType))
+      })
     // post to API
-    postRecords(access_token, data);
+    console.log('Posting records')
+    console.log(records)
+    postRecords(accessToken, records)
   }
 
   // map to records objects as needed
-  function parseKinesis(payload, avroType) {
+  function parseKinesis (payload, avroType) {
+    console.log('Parsing Kinesis')
     // decode base64
-    var buf = new Buffer(payload.kinesis.data, 'base64');
+    var buf = new Buffer(payload.kinesis.data, 'base64')
     // decode avro
-    var record = avroType.fromBuffer(buf);
-    return record;
+    var record = avroType.fromBuffer(buf)
+    return record
+  }
+
+  function addSource (record) {
+    console.log('Adding source')
+    record['nyplSource'] = 'sierra-nypl'
```

Comment:
Okay, I did this. In the future we can populate based on the upstream or add this as default. 🔮  When we know what the future holds...